staging/src/k8s.io/client-go/util/consistencydetector/data_consistency_detector.go

```diff
@@ -75,8 +74,8 @@ func CheckDataConsistency[T runtime.Object, U any](ctx context.Context, identity
 	sort.Sort(byUID(listItems))
 	sort.Sort(byUID(retrievedItems))
 
-	if !cmp.Equal(listItems, retrievedItems) {
-		klog.Infof("previously received data for %s is different than received by the standard list api call against etcd, diff: %v", identity, cmp.Diff(listItems, retrievedItems))
+	if !reflect.DeepEqual(listItems, retrievedItems) {
+		klog.Infof("previously received data for %s is different than received by the standard list api call against etcd, expected %+q but received %+q", identity, listItems, retrievedItems)
```

Comment:
I see support for diff [was discussed](https://github.com/kubernetes/kubernetes/issues/104821#issuecomment-2704307024).  I'd prefer we add diff support before we migrate call sites like this one where the diff output adds value.

Reply:
Yeah that's a fair ask, I was hoping to be able to merge this PR quickly but I understand having a clean diff would be best